### ISSUE TITLE
eval-all changed behavior in yq version 4.9.6

### ISSUE THREAD

**Issue by oguyotot** [mikefarah/yq#874]
**Describe the bug**

Not able to append to arrays the way I did it in yq 4.6.

```
yq -V
yq version 4.9.6 (Linux, binary)
```

**Input Yaml**

data1.yml:
```
data:
  array:
    - elem_name = "elem1"
      elem_value = "value1"
```

data2.yml:
```
data:
  array:
    - elem_name = "elem2"
      elem_value = "value2"
```

**Command**
The command you ran:
```
yq e '.data.array[0]' data2.yaml | yq eval-all 'select(fi==0).data.array += select(fi==1) | select(fi==0)' data1.yaml -
```
=> the goal is to append a yaml snippet from the standard input to an array as a new element (this is not a merge).

**Actual behavior**

The result is:
```
data:
  array:
    - elem_name = "elem1" elem_value = "value1"
```

**Expected behavior**

With yq v4.6, the result was:

```
data:
  array:
    - elem_name = "elem1" elem_value = "value1"
    - elem_name = "elem2" elem_value = "value2"
```

This is the result I am expecting, but I am not sure if the defect is in 4.9 or in 4.6. If 4.9 is producing the expected output, then how to get the same result than with 4.6?

**Comment by mikefarah** [comment]
Yep that's a bug, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.9.7 - also for what it's worth, you can do this in a single line:

```
yq ea 'select(fi==0).data.array += select(fi==1).data.array | select(fi==0)' data1.yaml data2.yaml
```

**Comment by oguyotot** [comment]
Thanks for tip, and for the fix of course!
